feeds/npm/npm.go

```diff
@@ -73,22 +53,56 @@ func fetchPackages() ([]*Package, error) {
 	if err != nil {
 		return nil, err
 	}
-	return rssResponse.Packages, nil
+	return rssResponse.PackageEvents, nil
 }
 
-// Gets the package version from the NPM.
-func fetchVersionInformation(packageName string) (string, error) {
-	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", versionURL, packageName))
+// Gets the package version & corresponding created date from NPM. Returns
+// a slice of {}Package.
+func createPackages(pkgTitle string, count int) ([]*Package, error) {
+	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", versionURL, pkgTitle))
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer resp.Body.Close()
-	v := &PackageVersion{}
-	err = json.NewDecoder(resp.Body).Decode(v)
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var jsonMap map[string]interface{}
+	err = json.Unmarshal(body, &jsonMap)
 	if err != nil {
-		return "", err
+		return nil, fmt.Errorf("%w : %v", errJSON, err)
 	}
-	return v.DistTags.Latest, nil
+
+	// The json string `time` contains tags in date order, oldest to newest.
+	versions, ok := jsonMap["time"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("%w : 'time' not found for package %s ",
+			errJSON, pkgTitle)
+	}
+
+	// Remove redudant entries in map, we're only interested in actual tags.
+	// If modified does not exist (new package), then the delete is no-op.
+	delete(versions, "created")
+	delete(versions, "modified")
+
+	// Create slice of Package{} to allow sorting of a slice, as maps
+	// are unordered.
+	tagSlice := []*Package{}
+	for tag, value := range versions {
+		date, err := time.Parse(time.RFC3339, value.(string))
+		if err != nil {
+			return nil, err
+		}
+		tagSlice = append(tagSlice, &Package{Title: pkgTitle, CreatedDate: date, Version: tag})
+	}
+
+	// Sort slice of tags into order of most recent.
+	sort.SliceStable(tagSlice, func(i, j int) bool {
```

Comment:
Do you want to throw in `feeds.SortPackages()` from [here](https://github.com/Qinusty/package-feeds/commit/2d500f0de1478fd16944666b5ae3c73282603551#diff-7c737d5ed17f49b3de045a735c3b8426c9b22a9fb2bd8832dc349e23ce0b8a0fR60-R65) since we're likely to be reusing this going forwards?

Reply:
Sure, I'll add a commit for it

Reply:
Ah, not so simple. At this point they're not the correct type `[]*feeds.Package{}`